lib/wallet/src/Cardano/Wallet/Shelley/Network/Discriminant.hs

```diff
@@ -1,23 +1,34 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE GADTs #-}
 {-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE UndecidableInstances #-}
 
 module Cardano.Wallet.Shelley.Network.Discriminant
     ( SomeNetworkDiscriminant (..)
     , networkDiscriminantToId
+    , discriminantNetwork
+    , EncodeAddress (..)
+    , EncodeStakeAddress (..)
+    , DecodeAddress (..)
+    , DecodeStakeAddress (..)
+    , HasNetworkId (..)
```

Comment:
🥳 I always wanted to see functionality related to the NetworkDiscrimant grouped in a single module, see [ADP-2134](https://input-output.atlassian.net/browse/ADP-2134).

That said, I think that `EncodeStakeAddress` should eventually end up in `Cardano.Wallet.Address` hierarchy, but this pull request is a perfectly good move.